src/Illuminate/Queue/QueueManager.php

```diff
@@ -295,6 +295,112 @@ public function setApplication($app)
         return $this;
     }
 
+    /**
+     * Pause a queue by name.
+     *
+     * @param  string  $connection
+     * @param  string  $queue
+     * @param  int|null  $ttl
+     * @return void
+     */
+    public function pause($connection, $queue, $ttl = null)
+    {
+        $cache = $this->app['cache']->store();
+
+        if ($ttl === null) {
```

Comment:
you do not need conditions here, put method already implements forever if ttl=null